pgx/src/pgx.ml

```diff
@@ -471,7 +471,8 @@ module type IO = Io_intf.S
 module type S = Pgx_intf.S
 
 module Make (Thread : IO) = struct
-  open Thread
+  module IO = Thread
+  open IO
```

Comment:
What's the reason for this change? It looks like it creates a bunch of downstream complexity (like the `Pgx_lwt.Io_intf` module) but doesn't seem to change anything else?

Reply:
It looks like we did this to accomidate Pgx_lwt, but I think we should keep our interface here simple and let Pgx_lwt do more complicated things to generate the interface if necessary. I'll try to make an example.